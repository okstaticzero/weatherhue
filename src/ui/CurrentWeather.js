import React, { Component } from 'react';


export class CurrentWeather extends Component {
  constructor(props){
    super(props)
    this.state={
      input:""
    }
  }

  onSubmit = (e) =>{
    this.props.getCurrentWeather(this.state.input)  
    e.preventDefault()
  }
  handleChange=(e) =>{
    this.setState({input:e.target.value})
  }
  render() {
    return (
      <div className="current-form">
        <h4 className="current-title">View current weather</h4>
        <form className="form-inline" onSubmit={this.onSubmit}>
          <div className="form-group">
            <input type="text" className="form-control" 
              onChange={this.handleChange} 
              value={this.state.input} 
              placeholder="Enter Address or zip" 
            />
            <button className="btn btn-default">GO</button>
          </div>  
        </form>
      </div>
    );
  }
}

export default CurrentWeather