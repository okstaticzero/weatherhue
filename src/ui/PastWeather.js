import React, { Component } from 'react';
import { DateRangePicker} from 'react-dates';
import { isInclusivelyBeforeDay } from 'react-dates'
import Select from 'react-select';
import moment from "moment"
import 'react-dates/lib/css/_datepicker.css';
import 'react-select/dist/react-select.css';

class PastWeather extends Component {
  constructor(props){
    super(props)
    this.state={
      interval:"21600",
      startDate: new moment().subtract(1, "day"),
      endDate : new moment()
    }
  }

  componentWillUpdate(){
    //console.log(this.state)
  }

  onSumbmit = (e) =>{
    e.preventDefault()
    let start = this.state.startDate
    let end = this.state.endDate
    if(!start || !end){
       alert("Please select both a start and end date")
       return
    }
    start = Number(start.format('X'))
    end = Number(end.format('X'))
    let int = Number(this.state.interval)
    console.log("start: "+ start)
    console.log("end: "+ end)
    if((end - start) < int){
      alert("Interval is greater than date range. Please change range or interval")
       return
    }
    this.props.setRangeInterval(start, end, int)
  }
  selectChange = (val) =>{
    this.setState({interval: val.value})
  }

  render() {
   var options = [
      { value: '3600', label: 'Every Hour' },
      { value: '7200', label: 'Every 2 Hour' },
      {value:  '21600', label: 'Every 6 Hour' },
      { value: '43200', label: 'Every 12 Hour' },
      { value: '86400', label: 'Every 24 Hour' },
      { value: '604800', label: 'Every 7 days' },
      { value: '2592000', label: 'Every 30 days' }
    ];

    
    return (
      <div className="past-form">
        <div className="date-pickers">
          <div className="date-wrapper">
            <form onSubmit={this.onSumbmit}>
            <h4 className="range-title">View weather by range / interval</h4>
            <h5 className="select-label">Date range:</h5>
              <DateRangePicker
                numberOfMonths={1}
                isOutsideRange={day => !isInclusivelyBeforeDay(day, moment())}
                startDate={this.state.startDate} // momentPropTypes.momentObj or null,
                endDate={this.state.endDate} // momentPropTypes.momentObj or null,
                onDatesChange={({ startDate, endDate }) => this.setState({ startDate, endDate })} // PropTypes.func.isRequired,
                focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                onFocusChange={focusedInput => this.setState({ focusedInput })}
              />
              <br/>
              <h5 className="select-label">Interval:</h5>
               <Select
                className="select-field"
                name="form-field-name"
                value={this.state.interval}
                options={options}
                onChange={this.selectChange}
                />
                <br/>
                <button className="btn btn-default">VIEW RESULTS</button>
            </form>  
          </div>
         
          
        </div>  
      </div>
    );
  }
}

export default PastWeather
