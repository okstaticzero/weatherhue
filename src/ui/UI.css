
.ui-container{
  display: inline-block;
  padding-bottom: 15px;
  border-bottom: 1px solid #CCC;
  text-align: left;
  padding-left: 30px;
  padding-right: 30px;
  background-color: rgba(255,255,255,.3);
  width:100%;
  color:#FFF;
}
@media screen and (min-width:800px){
  .ui-container{
    width:350px;
  }
}

.logo{
  margin: 40px 0px 30px 0px;
}
.form-control{
  width:222px;
  height:31px;
  border: 1px solid #CCC;
  outline:none;
  font-size:14px;
  padding-left: 10px;
  margin-right: 6px;
}
.btn.btn-default{
  background-color: #555;
  color:#FFF;
  border:0;
  font-size: 14px;
  padding: 8px 16px;
  cursor: pointer;
}
.current-form{
  padding-bottom: 20px;
  border-bottom: 1px solid #FFF;
}
.range-title{
  margin-top: 30px;
  font-size:19px;
  margin-bottom: 0;
}
.date-pickers{
  text-align: left;
  
}
.date-wrapper{
  width:100%;
  margin-bottom: 5px;
}
.date-wrapper h5{
  width:85px;
}
.date-wrapper h5, .rw-datetimepicker{
  display:inline-block;
  width: 100%
}
.select-field{
   width:295px;
}
.Select-control{
  border-radius:0;
}
.select-label{
  margin-bottom: 4px;
  font-size:13px;
}
.DateInput{
  padding: 2px
}
.DateRangePicker__picker{
  top:50px;
}

.DateInput--with-caret::after, .DateInput--with-caret::before{
  top:40px;
}
.Select-clear{
  display: none;
}

