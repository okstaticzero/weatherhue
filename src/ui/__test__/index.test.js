import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import UI from '../index.js'
import CurrentWeather from '../CurrentWeather.js'
import PastWeather from '../PastWeather.js'

it('should rener the UI component', () => {
  const wrapper = shallow(<UI />);
  expect(wrapper).toMatchSnapshot();
});

it('should rener the CurrentWeather component', () => {
  const wrapper = shallow(<CurrentWeather />);
  expect(wrapper).toMatchSnapshot();
});


it('should rener the PastWeather component', () => {
  const wrapper = shallow(<PastWeather />);
  expect(wrapper).toMatchSnapshot();
});