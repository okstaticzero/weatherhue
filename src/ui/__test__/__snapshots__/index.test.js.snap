// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rener the CurrentWeather component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="current-form"
>
    <h4
        className="current-title"
    >
        View current weather
    </h4>
    <form
        className="form-inline"
        onSubmit={[Function]}
    >
        <div
            className="form-group"
        >
            <input
                className="form-control"
                onChange={[Function]}
                placeholder="Enter Address or zip"
                type="text"
                value=""
            />
            <button
                className="btn btn-default"
            >
                GO
            </button>
        </div>
    </form>
</div>,
  "nodes": Array [
    <div
      className="current-form"
>
      <h4
            className="current-title"
      >
            View current weather
      </h4>
      <form
            className="form-inline"
            onSubmit={[Function]}
      >
            <div
                  className="form-group"
            >
                  <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="Enter Address or zip"
                        type="text"
                        value=""
                  />
                  <button
                        className="btn btn-default"
                  >
                        GO
                  </button>
            </div>
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CurrentWeather />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CurrentWeather {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "onSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "input": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="current-form"
>
          <h4
                    className="current-title"
          >
                    View current weather
          </h4>
          <form
                    className="form-inline"
                    onSubmit={[Function]}
          >
                    <div
                              className="form-group"
                    >
                              <input
                                        className="form-control"
                                        onChange={[Function]}
                                        placeholder="Enter Address or zip"
                                        type="text"
                                        value=""
                              />
                              <button
                                        className="btn btn-default"
                              >
                                        GO
                              </button>
                    </div>
          </form>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="current-form"
>
          <h4
                    className="current-title"
          >
                    View current weather
          </h4>
          <form
                    className="form-inline"
                    onSubmit={[Function]}
          >
                    <div
                              className="form-group"
                    >
                              <input
                                        className="form-control"
                                        onChange={[Function]}
                                        placeholder="Enter Address or zip"
                                        type="text"
                                        value=""
                              />
                              <button
                                        className="btn btn-default"
                              >
                                        GO
                              </button>
                    </div>
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CurrentWeather />,
}
`;

exports[`should rener the UI component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="ui-container"
>
    <img
        alt="Logo"
        className="logo"
        src="logo.png"
    />
    <CurrentWeather
        getCurrentWeather={undefined}
    />
    <PastWeather
        setRangeInterval={undefined}
        showAlert={undefined}
    />
</div>,
  "nodes": Array [
    <div
      className="ui-container"
>
      <img
            alt="Logo"
            className="logo"
            src="logo.png"
      />
      <CurrentWeather
            getCurrentWeather={undefined}
      />
      <PastWeather
            setRangeInterval={undefined}
            showAlert={undefined}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UI />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UI {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "input": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="ui-container"
>
          <img
                    alt="Logo"
                    className="logo"
                    src="logo.png"
          />
          <CurrentWeather
                    getCurrentWeather={undefined}
          />
          <PastWeather
                    setRangeInterval={undefined}
                    showAlert={undefined}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="ui-container"
>
          <img
                    alt="Logo"
                    className="logo"
                    src="logo.png"
          />
          <CurrentWeather
                    getCurrentWeather={undefined}
          />
          <PastWeather
                    setRangeInterval={undefined}
                    showAlert={undefined}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UI />,
}
`;
